pkg/administration/events/queue.go

```diff
@@ -0,0 +1,66 @@
+package events
```

Comment:
Would be good to add a metric to this so that we can track queue size?

Reply:
Had to do a bit of refactoring to allow metrics being included due to cyclic import: now the `logging` package only relies on the interface and doesn't call `Singleton` anymore (this is anyways what I wanted to in a follow-up).